dotCMS/src/main/java/com/dotmarketing/portlets/variant/business/VariantFactoryImpl.java

```diff
@@ -0,0 +1,121 @@
+     * @param variant
+     *
+     * @return
+     * @throws DotDataException
+     */
+    @Override
+    public Variant save(final Variant variant) throws DotDataException {
+
+        DotPreconditions.checkNotNull(variant.getName(), IllegalArgumentException.class,
+                "Name must not be null");
+        final String identifier = getId(variant);
+
+        new DotConnect().setSQL(VARIANT_INSERT_QUERY)
+                .addParam(identifier)
+                .addParam(variant.getName())
+                .addParam(variant.isArchived())
+                .loadResult();
+
+        return new Variant(identifier, variant.getName(), false);
+    }
+
+    private String getId(final Variant variant) {
+
+        final String deterministicID = DigestUtils.sha256Hex(variant.getName());
+
+        final Optional<Variant> variantFromDataBase;
+        try {
+            variantFromDataBase = get(deterministicID);
+
+            if (variantFromDataBase.isPresent()) {
+                return UUIDGenerator.generateUuid();
+            } else {
+                return deterministicID;
+            }
+        } catch (DotDataException e) {
+            return UUIDGenerator.generateUuid();
+        }
+    }
+
+    /**
+     * Implementation for {@link VariantFactory#update(Variant)}
+     * @param variant
+     *
+     * @throws DotDataException
+     */
+    @Override
+    public void update(Variant variant) throws DotDataException {
+        new DotConnect().setSQL(VARIANT_UPDATE_QUERY)
+                .addParam(variant.getName())
+                .addParam(variant.isArchived())
+                .addParam(variant.getIdentifier())
+                .loadResult();
+    }
+
+    @Override
+    public void delete(final String id) throws DotDataException {
+        new DotConnect().setSQL(VARIANT_DELETE_QUERY)
+                .addParam(id)
+                .loadResult();
+    }
+
+    @Override
+    public Optional<Variant> get(final String identifier) throws DotDataException {
+        final ArrayList loadResults = new DotConnect().setSQL(VARIANT_SELECT_QUERY)
+                .addParam(identifier)
+                .loadResults();
+
+        if (!loadResults.isEmpty()) {
+            final Map resultMap = (Map) loadResults.get(0);
+            return Optional.of(createVariant(resultMap));
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    public Optional<Variant> getByName(final String name) throws DotDataException {
+        final ArrayList loadResults = new DotConnect().setSQL(VARIANT_SELECT_BY_NAME_QUERY)
+                .addParam(name)
+                .loadResults();
+
+        if (!loadResults.isEmpty()) {
+            final Map resultMap = (Map) loadResults.get(0);
+            return Optional.of(createVariant(resultMap));
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    @NotNull
+    private Variant createVariant(Map resultMap) {
```

Comment:
This should be a transformer

Reply:
this method was removed

https://github.com/dotCMS/core/pull/22839/commits/8aaaf9e8f7d06c71edfd820737a9fcabc12d94ed#diff-a7d5f80eff1877d7090a757090861a8e6a45e62854b5e579eb7548089aabc4feL113

Reply:
approved